vlib/v/gen/c/testdata/return_optional_or_block.vv

```diff
@@ -0,0 +1,20 @@
+fn find_startswith_string(a []string, search string) ?string {
+	for s in a {
+		if s.starts_with(search) {
+			return s
+		}
+	}
+	return none
+}
+
+fn find_any_startswith_string(a []string, b []string, search string) ?string {
+	// cannot convert 'struct _option_string' to 'struct string'
+	// V wants the or {} block to return a string, but find_startswith_string returns ?string
+ 	return find_startswith_string(a, search) or { find_startswith_string(b, search) }
+}
+
+fn main() {
```

Comment:
```suggestion
fn test_main() {
```
and rename the file to `vlib/v/tests/return_optional_or_block_test.v` .

The pair of .vv/.out files is slower to test (they are run by `coutput_test.v`, which in turn tests them sequentially, while `v test .` runs the _test.v files in a thread pool.
`coutput_test.v` also relies on reading the generated output, and comparing it to a prerecorded known good value from the .out file, while the _test.v files are expected to assert themselves, and just exit with != 0, when there are failures.

For a program, that would have produced a cgen error on master, but will compile and run fine in the PR that fixes it, a _test.v is cheaper.

Reply:
OK thanks for the explanation. The CI found an issue that I am still looking at and I will update the new test.